### ISSUE TITLE
Regent: inadmissible OpenMP statement

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#697]
The following code:

```
import "regent"
local struct Particles_columns {
  cell : int3d;
  temperature : double;
}
local struct Fluid_columns {
  temperature : double;
}
__demand(__parallel)
task foo(Fluid : region(ispace(int3d), Fluid_columns),
         Particles : region(ispace(int1d), Particles_columns))
where
  reads(Particles.cell),
  writes(Particles.temperature),
  reads(Fluid.temperature)
do
  __demand(__openmp)
  for p in Particles do
    var cell = Particles[p].cell
    Particles[p].temperature = Fluid[cell].temperature
  end
end
```

when passed to:

```
regent.py test.rg -fopenmp 1 -fbounds-checks 1 -fbounds-checks-targets some_other_task
```

fails with:

```
:0: OpenMP code generation failed: found an inadmissible statement
```

Note that it's sufficient for bounds checks to be enabled, even if they don't apply to any task.

**Comment by magnatelee** [comment]
Should be fixed by 165522b4.

**Comment by manopapad** [comment]
Fixed
